ldk/javascript/examples/self-test-loop/src/tests/whisper/index.ts

```diff
@@ -430,6 +432,110 @@ export const testListPairWithCopyableLabel = (): Promise<boolean> =>
     }, 5000);
   });
 
+export const testMarkdownWithCopyableBody = (): Promise<boolean> =>
+  new Promise(async (resolve, reject) => {
+    const copyableText = '**Click me** to copy the Markdown value text';
+    const expectedCopiedText = `<p><strong>Click me</strong> to copy the Markdown value text</p>`;
+
+    // reset clipboard value
+    await clipboard.write('');
+
+    const createdWhisper = await whisper.create({
+      label: 'Copyable Markdown Test',
+      onClose: () => {
+        console.debug('closed');
+      },
+      components: [
+        {
+          type: WhisperComponentType.Markdown,
+          copyable: MarkdownWhisperCopyMode.Body,
+          body: copyableText,
+        },
+      ],
+    });
+
+    setTimeout(async () => {
+      const response = await clipboard.read();
+      if (response === expectedCopiedText) {
+        createdWhisper.close(() => {
+          // do nothing.
+        });
+        resolve(true);
+      } else {
+        reject(new Error('Incorrect value detected'));
+      }
+    }, 5000);
+  });
+
+export const testMessageWithCopyableBody = (): Promise<boolean> =>
+  new Promise(async (resolve, reject) => {
+    const copyableText = 'Click me to copy the Message value text';
+
+    // reset clipboard value
+    await clipboard.write('');
+
+    const createdWhisper = await whisper.create({
+      label: 'Copyable Markdown Test',
+      onClose: () => {
+        console.debug('closed');
+      },
+      components: [
+        {
+          type: WhisperComponentType.Message,
+          copyable: MessageWhisperCopyMode.Body,
+          body: copyableText,
+        },
+      ],
+    });
+
+    setTimeout(async () => {
+      const response = await clipboard.read();
```

Comment:
why not use clipboard.listen() instead. So you could avoid a set 5 sec delay?

Reply:
Good idea.

Reply:
Currently, the listen callback is not invoked when the electron app copies values to the clipboard. We will save this for a future user story if it is an issue.